include/uhr_func.hpp

```diff
@@ -533,105 +533,491 @@ static void zahlen(const char d1, const char d2) {
+        usedUhrType->show(h_fuenf);
         break;
     case 18:
-        uhrzeit |= ((uint32_t)1 << H_SECHS);
+        usedUhrType->show(h_sechs);
         break;
     case 19:
-        uhrzeit |= ((uint32_t)1 << H_SIEBEN);
+        usedUhrType->show(h_sieben);
         break;
     case 20:
-        uhrzeit |= ((uint32_t)1 << H_ACHT);
+        usedUhrType->show(h_acht);
         break;
     case 21:
-        uhrzeit |= ((uint32_t)1 << H_NEUN);
+        usedUhrType->show(h_neun);
         break;
     case 22:
-        uhrzeit |= ((uint32_t)1 << H_ZEHN);
+        usedUhrType->show(h_zehn);
         break;
     case 23:
-        uhrzeit |= ((uint32_t)1 << H_ELF);
+        usedUhrType->show(h_elf);
         break;
     case 24:
-        uhrzeit |= ((uint32_t)1 << H_ZWOELF);
+        usedUhrType->show(h_zwoelf);
         break;
     default:
         break;
     }
 }
+
 //------------------------------------------------------------------------------
 
 bool definedAndHasDreiviertel() {
-    return (G.Sprachvariation[ItIs45] == 1 &&
-            G.UhrtypeDef == Uhr_114_Alternative) ||
-           (G.Sprachvariation[ItIs45] == 1 && G.UhrtypeDef == Uhr_114_2Clock);
+    if (G.Sprachvariation[ItIs45] == 1) {
+        if (G.UhrtypeDef == Uhr_114_Alternative ||
+            G.UhrtypeDef == Uhr_114_2Clock || G.UhrtypeDef == Uhr_291) {
+            return true;
+        }
+    } else {
+        return false;
+    }
 }
 
 //------------------------------------------------------------------------------
 
-static void set_uhrzeit() {
-    uhrzeit = 0;
+void show_minuten(uint8_t min) {
+    if (G.zeige_min > 0) {
+        // Minuten / Sekunden-Animation
+        // Minute (1-4)  ermitteln
+        while (min > 4) {
+            min -= 5;
+        }
+
+        if (min > 0) {
+            Word_array[usedUhrType->getMinArr(G.zeige_min - 1, 0)] =
+                usedUhrType->getMinArr(G.zeige_min - 1, 0);
+        }
+        if (min > 1) {
+            Word_array[usedUhrType->getMinArr(G.zeige_min - 1, 1)] =
+                usedUhrType->getMinArr(G.zeige_min - 1, 1);
+        }
+        if (min > 2) {
+            Word_array[usedUhrType->getMinArr(G.zeige_min - 1, 2)] =
+                usedUhrType->getMinArr(G.zeige_min - 1, 2);
+        }
+        if (min > 3) {
+            Word_array[usedUhrType->getMinArr(G.zeige_min - 1, 3)] =
+                usedUhrType->getMinArr(G.zeige_min - 1, 3);
+        }
+    }
+}
 
+//------------------------------------------------------------------------------
+
+void set_minute(uint8_t min, uint8_t &offsetH, uint8_t &voll) {
+    if (G.UhrtypeDef != Uhr_291) {
+        show_minuten(min);
+        min /= 5;
+        min *= 5;
+    }
+    switch (min) {
+    case 0: // volle Stunde
+        usedUhrType->show(uhr);
+        voll = 1;
+        break;
+    case 1:
+        usedUhrType->show(m_eine);
+        usedUhrType->show(minute);
+        usedUhrType->show(nach);
+        break;
+    case 2:
```

Comment:
Vielleicht können wir uns hier auch viele Fälle sparen, wenn wir sicherstellen, dass die ganzen Minuten IDs (`m_*`) aufeinander folgend definiert sind damit wir in dem `switch` einfach sowas sagen können wie:
```
case 2:
case 3:
case 4:
    usedUhrType->show(m_start + min);
    usedUhrType->show(minuten);
    usedUhrType->show(nach);
```

Reply:
Ich habe das mal mit dem Commit https://github.com/ESPWortuhr/Wortuhr/pull/65/commits/c8e0efb945684667ade1f14ca4efe539a82e4291 angepasst.